EIPS/eip-draft_grant_standard.md

```diff
@@ -26,69 +26,282 @@ The current process for grants in the Ethereum ecosystem is for membership organ
+}
+```
+
+### Getter Methods
+
+#### manager
+Returns the multisig, EOA, or other address responsible for managing the grant. 
+```
+function manager() public view returns(address)
+```
+
+#### currency
+Returns the grant currency. If null, amount is in wei, otherwise address of ERC20-compliant (or other currency) contract.
+```
+function currency() public view returns(address)
+```
+
+#### targetFunding
+Returns the grant funding threshold required to begin releasing funds.
+```
+function targetFunding() public view returns(uint256)
+```
+
+#### fundingExpiration
+Returns the date after which funding must be complete.
+```
+function fundingExpiration() public view returns(uint256)
+```
+
+#### contractExpiration
+Returns the date after which payouts must be complete or anyone can trigger refunds.
+```
+function contractExpiration() public view returns(uint256)
+```
+
+#### grantCancelled
+Returns a flag to indicate whether or not grant is cancelled.
+```
+function grantCancelled() public view returns(bool)
+```
+
+#### grantees
+Returns Grantee.
+```
+function grantees(address) public view returns(Grantee)
+```
+
+#### donors
+Returns Donor.
+```
+function donors(address) public view returns(Donor)
+```
+
+#### getAvailableBalance
+Get available grant balance.
+```
+function getAvailableBalance() public view returns(uint256);
+```
+
+#### totalFunding
+Cumulative funding donated by donors.
+```
+function totalFunding() public view returns(uint256);
+```
+
+#### totalPayed
+Cumulative funding payed to grantees.
+```
+function totalPayed() public view returns(uint256);
+```
+
+#### totalRefunded
+OPTIONAL: If refunds permitted, cumulative funding refunded to donors.
+```
+function totalRefunded() public view returns(uint256);
+```
+
+#### pendingPayments
+OPTIONAL: If pull payments used, payments approved to grantees but not yet withdrawn.
+```
+function totalRefunded() public view returns(uint256);
+```
+
+
+
+#### canFund
+Funding status check. true if can fund grant.
+```
+function canFund() public view returns(bool);
+```
+
+### Initialization Methods
+
+#### constructor
+
+Grants are initialized on creation using the constructor. 
+
+```{sol}
+constructor(
+    address[] memory _grantees,
+    uint256[] memory _amounts,
+    address _manager,
+    address _currency,
+    uint256 _targetFunding,
+    uint256 _fundingExpiration,
+    uint256 _contractExpiration
+)
+```
+
+### State Modifying Methods
+
+#### fund
+Fund a grant proposal. `value` in WEI or GRAINS to fund.
+
+NOTE: This method is not `payable`. When funding with Ether, use fallback function to dispatch the `fund` method. 
+```
+function fund(uint256 value) public returns (bool);
+```
+
+#### approvePayout
+OPTIONAL: If managed by a `manager` this method approves payment to a grantee. If using push payouts, will also send payment.
+
+OPTIONAL ARG: Pay to a single `Grantee` when `grantee` address specified otherwise spit payout among `Grantee`s.
```

Comment:
This seems a bit iffy. Wouldn't it be simpler to push complexity out by enforcing one grantee? That grantee can be another smart contract which subdivides the funds further.

This extra optional argument might lead to some weird cases. Here's one of them: what if I want to fund a grantee who already has reached full funding? Would that go towards the next grantee? Would it be split up amongst the other grantees equally? Would it result in an error?

Reply:
I agree, it is more complex if we don't indicate who the grantee is (if one of the grantees is fully paid up the expected behaviour would be to make the tx fail, which involves more logic to do the check). The reference implementation does not leave out the optional arg so it mandates specifying which grantee is getting paid.

The logic for making it optional was to add some flexibility like the refund.

I'm game to remove the optional qualifier.

Reply:
Yep, let's remove. That'll encourage contract composition and remove special cases from this standard 👍 

I'd also make the argument that if people want the ability to fund multiple people independently, then each grantee should have its own grant contract.